protocol/binary_test.go

```diff
@@ -68,6 +75,18 @@ func checkEncodeDecode(t *testing.T, typ wire.Type, tests []encodeDecodeTest) {
 	}
 }
 
+func checkStreamEncode(t *testing.T, val testValue, expected []byte) {
+	var buff bytes.Buffer
+
+	// Encode with Streaming protocol
+	w := binary.BorrowStreamWriter(&buff)
+	err := val.StreamEncode(w)
+	require.NoError(t, err)
```

Comment:
The pattern under `StreamEncode` feels a bit weird...

Should a `*testing.T` be plumbed through?  Since the error checking is done here, the test failure would mark this line as errant making debugging the test difficult (it's unclear exactly where the error is since all of the tested calls are hidden away).

Alternatively, could you pass the specific function call in, e.g., `w.WriteInt32` to `checkStreamEncode` and execute that way?  This should be okay since a new `BorrowStreamWriter` is declared for every sub-test execution.

Reply:
Decide to remove StreamEncode for testing in general. Once you embed the StreamWriter into Writer, you test the code for the streaming implementation anyway.